sdk/core/azure-core/azure/core/rest/_helpers.py

```diff
@@ -285,22 +280,31 @@ def from_pipeline_transport_request_helper(request_class, pipeline_transport_req
     )
 
 def get_charset_encoding(response):
+    # type: (...) -> Optional[str]
     content_type = response.headers.get("Content-Type")
 
     if not content_type:
         return None
     _, params = cgi.parse_header(content_type)
     encoding = params.get('charset') # -> utf-8
-    if encoding is None:
-        if content_type in ("application/json", "application/rdap+json"):
-            # RFC 7159 states that the default encoding is UTF-8.
-            # RFC 7483 defines application/rdap+json
-            encoding = "utf-8"
-        else:
-            try:
-                encoding = chardet.detect(response.content)["encoding"]
-            except ResponseNotReadError:
-                pass
     if encoding is None or not lookup_encoding(encoding):
         return None
```

Comment:
We want to default to "utf-8"("utf-8-sig"), right?

Reply:
After talking with @lmazuel, we don't default the value of `self.encoding` to `utf-8-sig`. We do that when we're deserializing for the `text` property, which I've added code for in `def decode_to_text`